src/server/serve.rs

```diff
@@ -234,12 +207,81 @@ async fn start_dns_server(
     port: u32,
     rx: flume::Receiver<()>,
     no_proxy: bool,
+    nameservers: Vec<ScopedIp>,
 ) -> Result<(), anyhow::Error> {
-    let mut server =
-        CoreDns::new(addr, port, name, backend, rx, no_proxy).context("new dns server")?;
+    let mut server = CoreDns::new(addr, port, name, backend, rx, no_proxy, nameservers);
     server.run().await.context("run dns server")
 }
 
+async fn read_config_and_spawn(
+    config_path: &str,
+    port: u32,
+    filter_search_domain: &str,
+    handles_v4: &mut ThreadHandleMap<Ipv4Addr>,
```

Comment:
So this is declared as a HashMap, but counts as a ThreadHandleMap? Fascinating.

Reply:
type inference is very good in rust
https://doc.rust-lang.org/rust-by-example/types/inference.html